src/msssim.jl

```diff
@@ -0,0 +1,83 @@
+"""
+Computes the MS-SSIM between two images.
+
+!!! info
+
+    MSSSIM, Multi-scale SSIM is defined only for gray images. RGB images are treated as 3d Gray
+    images. General `Color3` images are converted to RGB images first, you could
+    manually expand them using `channelview` if you don't want them converted to
+    RGB first.
+```
+
+# Reference
+
+[1] Wang, Z., Bovik, A. C., Sheikh, H. R., & Simoncelli, E. P. (2004). Image quality assessment: from error visibility to structural similarity. _IEEE transactions on image processing_, 13(4), 600-612.
+
+"""
+struct MSSSIM{N} <: FullReferenceIQI
+    kernel::AbstractArray{<:Real}
+    W::NTuple{N, <:Tuple} # number of scales inferred from length
+    function MSSSIM(kernel, W)
+        ndims(kernel) == 1 || throw(ArgumentError("only 1-d kernel is valid"))
+        issymetric(kernel) || @warn "MSSSIM kernel is assumed to be symmetric"
+        all(length.(W) .== 3) || throw(ArgumentError("(α, β, γ) required for all scales, instead it's $(W)"))
+        all([ all(x .>= 0) for x in W ]) || throw(ArgumentError("α, β, γ should be non-negative for all scales, instead it's $(W)"))
+        new{length(W)}(kernel, W)
+    end
+end
+
+# Weights for α, β, γ in [1]
+const MSSSIM_W = (0.0448, 0.2856, 0.3001, 0.2363, 0.1333) # α, β, γ for scales 1 through 5
+
+MSSSIM(kernel, W::NTuple{N, <:Real}) where N = MSSSIM(kernel, map(x->(x, x, x), W)) # shorthand for αᵢ=βᵢ=γᵢ for all scales i
+
+# USE SSIM_KERNEL as default kernel
+MSSSIM(kernel=SSIM_KERNEL) = MSSSIM(kernel, MSSSIM_W)
+
+assess_msssim(x, ref) = MSSSIM()(x, ref)
+
+const DOWNSAMPLE_FILTER = ones(2, 2)./4 # as per author's implementaion
+
+# SSIM does not allow for user specifying peakval and K, so we don't allow it here either
+function (iqi::MSSSIM)(x::GenericImage, ref::GenericImage)
```

Comment:
Does `MSSSIM` work for other `Color3` (e.g., `HSV`, `Lab`) images? If so it would be good to also have some tests for them.

Reply:
It does not work out of  the box for `Lab` and `HSV`, throws errors saying things like `zero(::Lab{Float64})` and
`*(::HSV{Float64}, ::Float64)` aren't defined. We could convert these images to `RGB` before conversion, something like:
```julia
if eltype(a) <: Color{T, 3} where T
    a = RGB.(a)
end
```
for both `img` and `ref`

Reply:
Sounds good to me. Given that these metrics are only meaningful in RGB color space, I believe it won't cause much trouble to make such an assumption; I've done the same for SSIM, PSNR, and all other distances defined in ImageDistances.

Just be careful, there is another `AbstractRGB` type `BGR` and doing this requires a memory reallocation in such case. In this case, given that it contributes only `0.1%` to the executing time, it's totally okay.